### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by nslater**

### CODE REVIEW

File: docs/appendices/data-types.txt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+The ``ResultSet`` will have one column with the inner array type and the name
+of the array field (``Array.getBaseType()``) and as many rows as there are
+elements in the array.
+
+Here's one example:
+
+FIXME: can an engineer please add explanatory comments to this block of code.
+please explain what is being done here. please also format the code in the most
+idiomatic yet reasonable way (I have given this a go myself) - please also add
+empty lines where it makes most sense
```
Comment (nslater) [inline docs/appendices/data-types.txt]: please review this FIXME

File: docs/appendices/data-types.txt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+When inserting arrays using a prepared statement, you must convert your array
+to a `java.sql.Array`_ by the use of ``createArrayOf()``. This function takes
+as its first argument, a CrateDB type as described above and as its second the
+array you want to convert.
+
+You can then use ``setArray()`` to set this converted array.
+
+FIXME: can an engineer please add explanatory comments to this block of code.
+please explain what is being done here. please also format the code in the most
+idiomatic yet reasonable way (I have given this a go myself)
```
Comment (nslater) [inline docs/appendices/data-types.txt]: please review this FIXME

File: docs/appendices/data-types.txt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+You can fetch them using ``ResultSet.getObject()`` and cast the result to
+``Map<String, Object>``. This ``Map`` will contain all nested columns defined in
+the object.
+
+.. CAUTION::
+
+   Objects can be ``null``.
+
+FIXIME: is the above right? which object? the resulting java object?
```
Comment (nslater) [inline docs/appendices/data-types.txt]: please review this FIXME

File: docs/appendices/data-types.txt [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+FIXME: can an engineer please add explanatory comments to this block of code.
+please explain what is being done here. please also format the code in the most
+idiomatic yet reasonable way (I have given this a go myself)
```
Comment (nslater) [inline docs/appendices/data-types.txt]: please review this FIXME

File: docs/appendices/data-types.txt (hunk 1 above)
Comment (andreidan) [reply docs/appendices/data-types.txt]: Ok, so the second part of the code below is incorrect so needs to be removed. 

The code block should be:

```
Statement statement = connection.createStatement();
ResultSet resultSet = statement.executeQuery(
    "SELECT array_field FROM my_table"
);
resultSet.first();
Array arrayField = resultSet.getArray("array_field");
Object[] arrayFieldValue = arrayFieldValue.getArray();
```

Comment (andreidan) [reply docs/appendices/data-types.txt]: As a personal preference, I'd remove "my_table" and replace it with a real life example (selecting tags from a blob post table or something similar, whatever would match our existing documentation pattern)

File: docs/appendices/data-types.txt (hunk 2 above)
Comment (andreidan) [reply docs/appendices/data-types.txt]: lgtm

File: docs/appendices/data-types.txt (hunk 4 above)
Comment (andreidan) [reply docs/appendices/data-types.txt]: Just a bit of formatting and a variable rename. The code bellow should be:

```
        Statement statement = connection.createStatement();
        ResultSet resultSet = statement.executeQuery(
                "SELECT object_field FROM my_table"
        );
        resultSet.first();

        Map<String, Object> objectValue = (Map<String, Object>) resultSet.getObject("object_field");
        Object objectField = objectValue.get("nested_field");
```

File: docs/appendices/data-types.txt (hunk 3 above)
Comment (andreidan) [reply docs/appendices/data-types.txt]: The sql object type can hold null values. In which case ``ResultSet.getObject()`` will return ``null``.

**Comment by nslater** [comment]
@andreidan pushed a fixup. can you re-review please? ty!

**Pull request merged**